tests/spark_autologging/ml/test_pyspark_ml_autologging.py

```diff
@@ -245,3 +278,36 @@ def test_param_map_captures_wrapped_params(dataset_binomial):
     assert run_data.params == truncate_param_dict(
         stringify_dict_values(_get_instance_param_map(ova))
     )
+
+
+def test_pipeline(dataset_text):
+    mlflow.pyspark.ml.autolog()
+
+    tokenizer = Tokenizer(inputCol="text", outputCol="words")
+    hashingTF = HashingTF(inputCol=tokenizer.getOutputCol(), outputCol="features")
+    lr = LogisticRegression(maxIter=2, regParam=0.001)
+    pipeline = Pipeline(stages=[tokenizer, hashingTF, lr])
+    inner_pipeline = Pipeline(stages=[hashingTF, lr])
+    nested_pipeline = Pipeline(stages=[tokenizer, inner_pipeline])
+
+    assert _get_pipeline_stage_hierarchy(pipeline) == {
+        pipeline.uid: [tokenizer.uid, hashingTF.uid, lr.uid]
+    }
+    assert _get_pipeline_stage_hierarchy(nested_pipeline) == {
+        nested_pipeline.uid: [tokenizer.uid, {inner_pipeline.uid: [hashingTF.uid, lr.uid]}]
+    }
+
+    for estimator in [pipeline, nested_pipeline]:
+        with mlflow.start_run() as run:
+            model = estimator.fit(dataset_text)
+
+        run_id = run.info.run_id
+        run_data = get_run_data(run_id)
+        assert run_data.params == truncate_param_dict(
+            stringify_dict_values(_get_instance_param_map(estimator))
```

Comment:
@WeichenXu123 can we add some test coverage for `_get_instance_param_map` to ensure that it's producing a reasonable set of parameters? Otherwise, this test would pass even if the parameters returned by `_get_instance_param_map` were broken / incorrect.